capstone/src/main/webapp/js/autocomplete.js

```diff
@@ -0,0 +1,140 @@
+const inp = document.getElementById('search');
+const currentFocus = -1;
+/**
+ * @param {Object} autocompleteListElmt
+ * @return {bool}
+ * Adds the 'active' tag to an autocomplete elmt
+ */
+function addActiveTag(autocompleteListElmt) {
+  if (!autocompleteListElmt) {
+    return false;
+  }
+  removeActiveTag(autocompleteListElmt);
+  if (currentFocus >= autocompleteListElmt.length) {
+    currentFocus = 0;
+  } else if (currentFocus < 0) {
+    currentFocus = (autocompleteListElmt.length - 1);
+  }
+  autocompleteListElmt[currentFocus].classList.add('autocomplete-active');
+}
+
+/**
+ * @param {Object} autocompleteListElmt
+ * @return {void}
+ * Removes the 'active' tag from an autocomplete elmt
+ */
+function removeActiveTag(autocompleteListElmt) {
+  for (listElmt in autocompleteListElmt) {
+    if (true) {
+      listElmt.classList.remove('autocomplete-active');
+    }
+  }
+}
+
+/**
+ * @param {Object} elmnt
+ * @param {Object} inp
+ * @return {void}
+ * Closes dropdown autocomplete list(s)
+ */
+function closeAllLists(elmnt, inp) {
+  const autoItems = document.getElementsByClassName('autocomplete-items');
+  for (item in autoItems) {
+    if (elmnt != item && elmnt != inp) {
+      item.parentNode.removeChild(item);
+    }
+  }
+}
+
+/**
+ * @return{Object} arr
+ * Current placeholder until servlet is created
+ */
+function getListOfSuggestions() {
+  const arr = ['Hello', 'Hi', 'Howdy'];
+  return arr;
+}
+
+/**
+ * @param {Object} arr
+ * @param {string} val
+ * @return {Object} trimArr
+ * Identifies and returns relevant suggestions in the arr
+ */
+function getRelevantDataSuggestions(arr, val) {
+  const trimArr = [];
+  for (const arrElt of arr) {
+    if (arrElt.substr(0, val.length).toUpperCase() == val.toUpperCase()) {
+      trimArr.push(arrElt);
+    }
+  }
+  return trimArr;
+}
```

Comment:
OOC - shouldn't the server be making sure that relevant suggests are returned to the client?

We could also imagine a case where a client sends in "UCLA" but the response is the entire name. Wouldn't we end up filtering out such suggestions here?

Reply:
I think this is the fake server implementation so we can manually testing autocomplete.

Reply:
Actually i don't know. Cause getListOfSuggestions also looks like it could be the fake server.

Reply:
Yeah this is like a fake server implementation so we can test the autocomplete functionality.

Reply:
Is there a better way to approach this?

Reply:
ah ok...would you recommend doing that now or should I change that in a follow-up PR that will actually get the data from the servlet and connect wires?